### PR TITLE
remove announcement that was causing 1 way voice

### PR DESCRIPTION
**Pull request by smckinnon-bw**
the primary change is the commenting out of line 197 of server.ts to remove a SpeakSentence that was causing a re-invite.

the other changes were to remove IDE errors pertaining to a rule that says "no applying types to errors in a catch"

I think that I'll go with one reviewer since this repo allows it, and I'd like to get it to TAC and the customer.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -380,12 +380,17 @@ const createParticipant = async (tag: string): Promise<ParticipantInfo> => {
 const deleteSession = async () => {
   if (sessionId) {
     try {
-      await webRTCController.deleteSession(accountId, sessionId);
       console.log(`Deleted WebRTC session: ${sessionId} `);
       sessionId = "";
     } catch (e) {
+      let error: any = e;
```
Comment (psherk-bw) [inline src/server.ts:386]: you should be able to type cast this as `Error` so 

`let error: Error = e;`

File: src/server.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ app.post("/bridgeCallAnswered", async (req, res) => {
   });
 
   const resp = new Response();
-  resp.add(speak);
+  // resp.add(speak);  //this introduces a 1-way voice condition - TODO - fix the root cause and reinstate the annct
```
Comment (psherk-bw) [inline src/server.ts:197]: I would completely remove this line, and add a JIRA ticket or just a TODO comment with the work to be done

**Review by psherk-bw — COMMENTED** [review]
Couple of comments, also we should not be deleting the `package-lock.json` files. Those are good to have

File: src/server.ts (hunk 1 above)
Comment (smckinnon-bw) [reply src/server.ts:386]: good point - thanks.  I'll evolve it

File: src/server.ts (hunk 2 above)
Comment (smckinnon-bw) [reply src/server.ts:197]: wisdom !  thanks.

File: src/server.ts (hunk 1 above)
Comment (smckinnon-bw) [reply src/server.ts:386]: question on the package-lock.json comment.  package-lock.json changed because I suspect there are dependency version updates.  package.json and package-lock.json were not included in .gitignore.  Bottom line - I don't understand the issue.

Comment (psherk-bw) [reply src/server.ts:386]: Oh there were so many deletions I assumed it was deleted. Looks like it wasn't entirely deleted, but I suppose some things updated. The package-lock does get filled with the exact dependencies, and the dependencies of those dependencies with their corresponding versions "locked in" hard.

Comment (psherk-bw) [reply src/server.ts:386]: it looks fine to me! Just was curious why it deleted almost 20k lines

File: src/server.ts [hunk 3]
```diff
@@ -380,12 +380,17 @@ const createParticipant = async (tag: string): Promise<ParticipantInfo> => {
 const deleteSession = async () => {
   if (sessionId) {
     try {
-      await webRTCController.deleteSession(accountId, sessionId);
```
Comment (carrington-bw) [inline src/server.ts:383]: Dumb question, but why did you remove this line? It looks to me that the only code in the try block is resetting the session Id and a console.log()

File: src/server.ts [hunk 4]
```diff
@@ -380,12 +380,17 @@ const createParticipant = async (tag: string): Promise<ParticipantInfo> => {
...
       console.log(`Deleted WebRTC session: ${sessionId} `);
       sessionId = "";
     } catch (e) {
+      let error: any = e;
       console.log("failed to delete session", sessionId);
-      console.log("error", e.response.status, e.response.data, e.config.url);
+      console.log(
```
Comment (carrington-bw) [inline src/server.ts:388]: Does this look okay as displayed in the console?

File: src/server.ts (hunk 1 above)
Comment (smckinnon-bw) [reply src/server.ts]: I cleaned up the typing of the catch clause (e) parameter, and that also caused me to find and fix errors in the console.log outputs in one of the catch clauses, and to find and fix a missing call to actually delete the session.

File: src/server.ts (hunk 2 above)
Comment (smckinnon-bw) [reply src/server.ts]: SpeakSentence stuff removed, and ticket EDGE-1171 opened to replace it someday (taking bets ?)

File: src/server.ts (hunk 3 above)
Comment (smckinnon-bw) [reply src/server.ts]: that was a fat-finger mistake.

I sort of wondered where it went when I got to thinking "when does console.log throw an exception?"

It's back there now.

File: src/server.ts [hunk 5]
```diff
@@ -380,16 +375,18 @@ const createParticipant = async (tag: string): Promise<ParticipantInfo> => {
 const deleteSession = async () => {
   if (sessionId) {
     try {
+      await webRTCController.deleteSession(accountId, sessionId);
       console.log(`Deleted WebRTC session: ${sessionId} `);
       sessionId = "";
     } catch (e) {
-      let error: any = e;
+      let error: ApiResponse<void> = e as ApiResponse<void>;
```
Comment (psherk-bw) [inline src/server.ts:382]: Did this end up working (tested?), or is this supposed to be `ApiError`

Comment (psherk-bw) [reply src/server.ts:382]: might be worth doing something like 

```
if (e instanceof ApiError) {
        const error: ApiError = e
        // do logging
}
```

Comment (smckinnon-bw) [reply src/server.ts:382]: I tested it and it worked, although it exposed the fact that the error report for the delete participant in the original code would never have worked (and has been fixed)

File: src/server.ts (hunk 4 above)
Comment (smckinnon-bw) [reply src/server.ts:384]: not sure what 'OK ' looks like, but it is clear and comprehensive.  it exposes perhaps too much data.

Comment (carrington-bw) [reply src/server.ts:384]: Mainly wanting to see if it works and displays what you want it to

**Review by carrington-bw — APPROVED** [review]

**Review by psherk-bw — APPROVED** [review]

File: src/server.ts [hunk 6]
```diff
@@ -375,18 +375,17 @@ const createParticipant = async (tag: string): Promise<ParticipantInfo> => {
 const deleteSession = async () => {
   if (sessionId) {
     try {
-      await webRTCController.deleteSession(accountId, sessionId);
+      await webRTCController.deleteSession(accountId, "garbage");
```
Comment (psherk-bw) [inline src/server.ts:378]: is there a reason we put "garbage" in here?

**Review by carrington-bw — APPROVED** [review]

**Review by psherk-bw — APPROVED** [review]

File: src/server.ts (hunk 6 above)
Comment (smckinnon-bw) [reply src/server.ts]: my bad - that was for testing and I screwed up the commit/push.
I fixed it with the last commit.
